### PR TITLE
Modal Madness II: The Redirectioning

### LINKED ISSUE #74: Provide menu for selecting a game to join
Once #73 is implemented, it should be possible for the join game (as player and spectator) option to present a list of potential games to join. It should display at a minimum, the name of the hosting player.

### PR DESCRIPTION
**Pull request by acmorales**
CHANGES
-
- Closes #74 and #76 
- Improved the `JoinGameModal` by including list views for decks and available games
- Configured browser redirects on game host/join

NOTES
-
- `hostGameFunction()` should probably add a `uid` property to the created games (I manually added it to the existing games for this demo) #78 
- How are we going to identify games to join? Should you be able to name them when you host? Maybe we could show usernames for players currently in a game?

DEMO
-
Join game
joinGamev2

Redirects
redirect

### CODE REVIEW

File: client/src/components/GameArea.js [hunk 1]
```diff
@@ -189,4 +191,4 @@ GameArea.propTypes = {
   cards: PropTypes.array.isRequired,
 };
 
-export default GameArea;
+export default withAuthentication(GameArea);
```
Comment (eeeeeric) [inline client/src/components/GameArea.js]: I'm curious why we wrap in here rather than when another component imports the GameArea.

**Comment by eeeeeric** [comment]
Also FYI, if you put multiple issue numbers with a single "Closes", GitHub only auto closes the first one.

**Comment by eeeeeric** [comment]
> How are we going to identify games to join? Should you be able to name them when you host? Maybe we could show usernames for players currently in a game?

When we implement #73 and #78, it should be a little easier to ID games. In addition to the lifecycle status and uid, we should expose:
- when the game was started (so we can sort to show more recently created games first)
- the players currently in the game, as well as who the host is (we can show how many spots are left in the game and who the host is)

We could add a name to the game, but I'm not sure how valuable it will be (do I just name all my games 334?)

File: client/src/components/GameArea.js (hunk 1 above)
Comment (acmorales) [reply client/src/components/GameArea.js]: No real reason, it'd probably be better to wrap this where we import it actually. Will push a new commit

**Comment by acmorales** [comment]
I think that being able to show the host as well as the players currently in the game would be enough to identify games. So if we update the `hostGame()` and `joinGame()` functions to include/update this information in the game document, we can display that information

**Pull request merged**